Src/WindowsServer/WindowsServer.Shared/Implementation/AppServiceEnvVarMonitor.cs

```diff
@@ -0,0 +1,57 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Utility to monitor the value of environment variables which may change 
+    /// during the run of an application. Checks the environment variables 
+    /// intermittently.
+    /// </summary>
+    internal static class AppServiceEnvVarMonitor
+    {
+        internal static TimeSpan CheckInterval = TimeSpan.FromSeconds(30);
+        internal static DateTime LastCheckTime = DateTime.MinValue;
```

Comment:
Nit: `private` (unless you need to change it in unit tests.)

Reply:
`CheckInterval` can be `private`, `NextCheckTime` internal to allow tests to work - added a line comment to call this intention out.